var/spack/repos/builtin/packages/netcdf-c/package.py

```diff
@@ -89,11 +95,12 @@ class NetcdfC(AutotoolsPackage):
     #         description='Enable CDM Remote support')
 
     # The patch for 4.7.0 touches configure.ac. See force_autoreconf below.
-    depends_on("autoconf", type="build", when="@4.7.0,main")
-    depends_on("automake", type="build", when="@4.7.0,main")
-    depends_on("libtool", type="build", when="@4.7.0,main")
+    with when("build_system=autotools"):
+        depends_on("autoconf", type="build", when="@4.7.0,main")
+        depends_on("automake", type="build", when="@4.7.0,main")
+        depends_on("libtool", type="build", when="@4.7.0,main")
+        depends_on("m4", type="build")
```

Comment:
Are you sure, we don't need `m4` when building with `cmake` (e.g. [here](https://github.com/Unidata/netcdf-c/blob/43abd699e19db24e27e1e800086ff8142f3b07ad/CMakeLists.txt#L2016))?

Reply:
netcdf-c master (and any subsequent releases post `4.9.0`) will likely need this enabled yes, however for the time being netcdf-c builds perfectly fine without it as the files m4 is looking to generate are checked into source. I'll add the dep back for linux platforms, but as Spack does not currently support installing m4 for Windows, we won't add it as a dep for Windows.